src/analysisd/decoders/decoder.h

```diff
@@ -79,18 +81,37 @@ typedef struct dbsync_context_t {
  * list and to get the first osdecoder
  */
 void OS_CreateOSDecoderList(void);
-int OS_AddOSDecoder(OSDecoderInfo *pi, OSDecoderNode **pn_osdecodernode, OSDecoderNode **npn_osdecodernode);
+int OS_AddOSDecoder(OSDecoderInfo *pi, OSDecoderNode **pn_osdecodernode, OSDecoderNode **npn_osdecodernode, char **msg);
 OSDecoderNode *OS_GetFirstOSDecoder(const char *pname);
 int getDecoderfromlist(const char *name);
 char *GetGeoInfobyIP(char *ip_addr);
-int SetDecodeXML(void);
+/**
+ * @brief Add decoders of base modules to main list.
```

Comment:
Add decoders to main list --> Read decoder files and save them in the decoder list.
(On the next iterations, we will do this function receive as parameter a list which saves decoders).